nodejs/README.adoc

```diff
@@ -0,0 +1,341 @@
+
+.Verification
+ifdef::qs[]
+* Is the `countries` topic listed in the topics table?
+endif::[]
+ifndef::qs[]
+* Verify that the `countries` topic is listed in the topics table.
+endif::[]
+
+[id="proc-running-nodejs-example-application_{context}"]
+== Running the Node.js example application
+
+After you configure your Node.js application to connect to a Kafka instance and you create the required Kafka topic, you can run the Node.js application to start producing and consuming messages.
+
+In this task, you'll use the following components of the Node.js application to produce and consume messages:
+
+* A `producer-backend` component that generates random country names and sends these names to the Kafka topic.
+* A `consumer-backend` component that consumes the country names from the Kafka topic.
+
+.Prerequisites
+* You've configured the Node.js example application to connect to the Kafka instance.
+* You've created the `countries` example Kafka topic.
+
+.Procedure
+. On the command line, navigate to the `reactive-example` directory corresponding to the repository that you cloned.
++
+.Navigating to the reactive-example directory
+[source]
+----
+$ cd ~/reactive-example
+----
+
+. Check out the `kafkajs` branch of the repository.
++
+.Checking out the kafkajs branch
+[source]
+----
+$ git checkout kafkajs
+----
+
+. Run the shell script that sets environment variables for the bootstrap server endpoint and client credentials. These are required to connect to your Kafka instance.
++
+.Running the shell script
+[source]
+----
+$ source set-environment.sh
+----
+
+. Navigate to the directory for the consumer component. Use Node Package Manager (npm) to install the package for this component.
++
+.Installing the consumer component
+[source]
+----
+$ cd consumer-backend
+$ npm install
+----
+
+. Run the consumer component.
++
+.Running the consumer component
+[source]
+----
+$ node consumer.js
+----
++
+You should see the Node.js application start to run and connect to the Kafka instance. However, because you haven't run the producer component yet, the consumer has no country names to display.
++
+If the application fails to run, review the error log in the command-line window and address any problems. Also, review the steps in this quick start to ensure that the application and Kafka topic are configured correctly.
+
+. On your computer, open a second command-line window or tab. On the command line, navigate to the `reactive-example` directory corresponding to the repository that you cloned.
++
+.Navigating to the reactive-example directory
+[source]
+----
+$ cd ~/reactive-example
+----
+
+. Check out the `kafkajs` branch of the repository.
++
+.Checking out the kafkajs branch
+[source]
+----
+$ git checkout kafkajs
+----
+
+. Run the shell script that sets environment variables for the bootstrap server endpoint and client credentials.
++
+.Running the shell script
+[source]
+----
+$ source set-environment.sh
```

Comment:
This is antipattern for node.js.

Would it be possible to use something like https://www.npmjs.com/package/dotenv in the repo and remove those steps from guide?

Reply:
I did that to avoid having people having to export the variables multiple times while still going over what you need to do so that people would understand and can apply to whatever they will do in their environment.  I looks like using a .env file might be similar.  They key thing for me is that the person going through it knows what they need to set/use in their code.

Reply:
The nice thing about using .env would be that it should work the same across platforms so for that reason I think it's worth doing.

I'll see if I can find tomorrow to adjust for using .env.

Reply:
Thank you. .Env is used by Quarkus guide as well.

Reply:
@mhdawson  Let me know if you implement this change. Right now, we instruct users to create a shell script for Linux/macOS, or a batch file for Windows.

Reply:
@jbyrne-redhat will do, pretty busy but might find time on Friday , will let you know.

Reply:
From https://access.redhat.com/documentation/en-us/red_hat_openshift_streams_for_apache_kafka/1/guide/04827d87-ed92-4ffd-a126-11fa13348eba looks to me like the quarkus guide uses environment variables, but still think its a good idea to look at the .env approach for the Node.js guide.

Reply:
PR to update the kafka version to support dotenv: https://github.com/nodeshift-starters/reactive-example/pull/34. Working on node-rdkafka version.

Reply:
PR for node-rdkafka - https://github.com/nodeshift-starters/reactive-example/pull/35

Both PRs are landed, and with the suggestions I made to swtich to creating '.env' file I think I'm done.

@jbyrne-redhat let me know if you need anything else.